doc/rtd/topics/datasources/nocloud.rst

```diff
@@ -32,15 +32,84 @@ The permitted keys are:
 
 With ``ds=nocloud``, the ``seedfrom`` value must start with ``/`` or
 ``file://``.  With ``ds=nocloud-net``, the ``seedfrom`` value must start
-with ``http://`` or ``https://``.
+with ``http://`` or ``https://`` and end with a trailing ``/``.
```

Comment:
Is this true? Why do we need a trailing '/'?

If so, we should update https://github.com/canonical/cloud-init/blob/main/doc/examples/cloud-config-datasources.txt#L34 accordingly.

Reply:
This is true. Its caught me out before

Reply:
I've added a commit to this PR to auto-append the missing / if absent in seedfrom urls (which don't already contain a `%s`).